ftw/linkchecker/report_generating.py

```diff
@@ -1,40 +1,76 @@
+from Products.CMFPlone.utils import safe_unicode
+from ftw.linkchecker.command import broken_link
+from io import BytesIO
+from plone import api
+import re
 import xlsxwriter
 
-LABELS = [
-    [
-        'Internal/External',
-        'Origin',
-        'Destination',
-        'Status Code',
-        'Content Type',
-        'Response Time',
-        'Header Location',
-        'Error Message'
-    ]
-]
+LABELS = broken_link.BrokenLink()
+LABELS.is_internal = 'Internal/External'
+LABELS.link_origin = 'Origin'
+LABELS.link_target = 'Destination'
+LABELS.status_code = 'Status Code'
+LABELS.content_type = 'Content Type'
+LABELS.response_time = 'Response Time'
+LABELS.header_location = 'Header Location'
+LABELS.error_message = 'Error Message'
+LABELS = [LABELS]
+
+NUMBER_OF_LABELS = 8
```

Comment:
That's `len(LABELS.table_attrs)` ? This way you can avoid that.

Reply:
That's definitely the better way to do that. I'm adapting that.